src/mlj_docstrings.jl

```diff
@@ -0,0 +1,146 @@
+
+const CATBOOST_DOCS_LINK = "https://catboost.ai/en/docs/"
+const CATBOOST_PARAMS_DOCS_LINK = "https://catboost.ai/en/docs/references/training-parameters/"
+
+"""
+$(MMI.doc_header(CatBoostRegressor))
+
+# Training data
+
+In MLJ or MLJBase, bind an instance `model` to data with
+
+    mach = machine(model, X, y)
+
+where
+
+- `X`: any table of input features (eg, a `DataFrame`) whose columns
+  each have one of the following element scitypes: `Continuous`,
+  `Count`, `Finite`, `Textual`; check column scitypes with `schema(X)`.
+  `Textual` columns will be passed to catboost as `text_features`,
+  `Multiclass` columns will be passed to catboost as `cat_features`, and
+  `OrderedFactor` columns will be converted to integers.
+
+- `y`: the target, which can be any `AbstractVector` whose element
+  scitype is `Finite`; check the scitype with `scitype(y)`
+
+Train the machine with `fit!(mach, rows=...)`.
+
+
+# Hyper-parameters
+#TODO: Document hyper-parameters
```

Comment:
I think given the number of parameters and the associated maintenance burden, it suffices to refer to the catboost.ai docs for the details. If there are hyperparameter values which would differ (e.g., symbol expected instead of string, or the value is an instance of some CatBoost.jl custom type) then these should be noted. But maybe this has been avoided?

Reply:
One option is to interpolate in the python docs if there's some appropriate ones. That's what we did here: https://github.com/beacon-biosignals/CatBoost.jl/blob/9cab2d6ebce3cf205923a26140b4bda2443907b3/src/CatBoost.jl#L44-L46

Not perfect but I think it's helpful still